flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -615,11 +615,10 @@ private void stopTaskExecutorServices() throws Exception {
 					CompletableFuture.runAsync(
 						() -> {
 							try {
-								if (!networkEnvironment.updatePartitionInfo(executionAttemptID, partitionInfo)) {
+								if (!networkEnvironment.updatePartitionInfo(getResourceID(), executionAttemptID, partitionInfo)) {
 									log.debug(
-										"Discard update for input gate partition {} of result {} in task {}. " +
+										"Discard update for input gate partition of result {} in task {}. " +
 											"The partition is no longer available.",
-										partitionInfo.getInputChannelDeploymentDescriptor().getConsumedPartitionId(),
```

Comment:
we could still use partitionInfo.getShuffleDescriptor.getResultPartitionID here?